operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/DefaultContext.java

```diff
@@ -2,16 +2,39 @@
 
 import java.util.Optional;
 
-public class DefaultContext implements Context {
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.processing.Controller;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+import io.javaoperatorsdk.operator.processing.event.source.EventSourceRegistry;
+
+public class DefaultContext<P extends HasMetadata> implements Context<P> {
 
   private final RetryInfo retryInfo;
+  private final Controller<P> controller;
+  private final P primaryResource;
 
-  public DefaultContext(RetryInfo retryInfo) {
+  public DefaultContext(RetryInfo retryInfo, Controller<P> controller, P primaryResource) {
```

Comment:
Should this go to primary context? Cannot be an implementation of `Reconciler` that prodives these? 
To have the current default  APIs unchanged? Wouldn't that be possible?